code-challenges.test.js

```diff
@@ -15,37 +15,126 @@
 // a) Create a test with an expect statement using the variable provided.
 // HINT: Check out this resource: https://jestjs.io/docs/expect#expectarraycontainingarray
 
-const colors1 = ["purple", "blue", "green", "yellow", "pink"]
+describe("shuffleArray", () => {
+  const colors1 = ["purple", "blue", "green", "yellow", "pink"]
+  const colors2 = [
+    "chartreuse",
+    "indigo",
+    "periwinkle",
+    "ochre",
+    "aquamarine",
+    "saffron",
+  ]
+  it("returns a new array that has removed the first item from the original array and shuffled the rest of the items.", () => {
+    expect(shuffleArray(colors1)).toEqual(expect.arrayContaining(colors1))
+    expect(shuffleArray(colors2)).toEqual(expect.arrayContaining(colors2))
+  })
+})
+
 // Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
-const colors2 = [
-  "chartreuse",
-  "indigo",
-  "periwinkle",
-  "ochre",
-  "aquamarine",
-  "saffron"
-]
 // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
 // b) Create the function that makes the test pass.
+//Puesdo C:
+// Create a function called shuffle array to match the test.
+// Give the funciton one parameter 'array' too match the data to bee arguments.
+//Start the function by designating a new variable called remove first that will take out the first item from the array. 
+// Store indside the variable the parameter array and use the .shift() method to remove the first item.
+// Next, create a for loop that will iterate through the array and set the the index to equal the array.length - 1. 
+// Next, set i > 0 and we want i to decrement so set i --
+// inside the for loop create a new variable that will hold the logic for the Fisher-Yates algorithm. 
+// This will give a random index within the remaining elements of the array, and then swap the curren element with the random one. 
+// Last return the array that has removed the first value and shuffled the rest. 
+
+const shuffleArray = (array) => {
+  let removeFirst = array.shift()
+  for (let i = array.length - 1; i > 0; i--) {
+    let newArray = Math.floor(Math.random() * (i + 1))
+    let a = array[newArray]
+    let b = array[i]
+    array[i] = a
+    array[newArray] = b
+  }
+  return array
+}
+
```

Comment:
It looks like you created a variable called removeFirst that you are not using. Since shift is a mutator you actually don't need to store it in a variable.